controllers/secretscope_controller_databricks.go

```diff
@@ -152,25 +152,66 @@ func (r *SecretScopeReconciler) submitACLs(instance *databricksv1alpha1.SecretSc
 	return nil
 }
 
-func (r *SecretScopeReconciler) submit(instance *databricksv1alpha1.SecretScope) error {
+// checkCluster checks if Databricks cluster supports ACLs, and checks if secret scope exists.
+func (r *SecretScopeReconciler) checkCluster(instance *databricksv1alpha1.SecretScope) error {
 	scope := instance.ObjectMeta.Name
 	initialManagePrincipal := instance.Spec.InitialManagePrincipal
 
+	// try create secret scope to see if exists or not.
 	err := r.APIClient.Secrets().CreateSecretScope(scope, initialManagePrincipal)
 	if err != nil {
 		return err
 	}
 
-	err = r.submitSecrets(instance)
-	if err != nil {
-		return err
+	// try to list ACLs to see if cluster supports ACL.
+	if instance.Spec.SecretScopeACLs != nil {
+		if _, err = r.APIClient.Secrets().ListSecretACLs(scope); err != nil {
+			if er := r.APIClient.Secrets().DeleteSecretScope(scope); er != nil {
+				return fmt.Errorf("%v\n%v", err, er)
+			}
+			return err
+		}
 	}
 
-	err = r.submitACLs(instance)
+	return nil
+}
+
+// checkSecrets checks if referenced secret is present in k8s or not.
+func (r *SecretScopeReconciler) checkSecrets(instance *databricksv1alpha1.SecretScope) error {
+	scope := instance.ObjectMeta.Name
+	namespace := instance.Namespace
+
+	// if secret in cluster is reference, see if secret exists.
+	// if secret does not exist, try delete secret scope.
+	for _, secret := range instance.Spec.SecretScopeSecrets {
+		if secret.ValueFrom != nil {
+			if _, err := r.getSecretValueFrom(namespace, secret); err != nil {
+				if er := r.APIClient.Secrets().DeleteSecretScope(scope); er != nil {
```

Comment:
if we use get secret scope perhaps we don't need to delete the secrete scope here

Reply:
The reason for deleting the secret scope is because what happens is if the secret in the cluster doesn't exist, the config will be put back onto the reconcile loop. Next time it runs, it will check if the secret scope exists in databricks and because it was created the previous run, `verifyWorkspace` will cause it to fail and thus wont check again for the secret in cluster. 

Happy for suggestions.

Reply:
I've cleaned it up so that it doesn't have to delete the secret scope any longer.